### ISSUE TITLE
playBinary - gradle 4.2 play 2.6 - using the wrong main class

### ISSUE THREAD

**Issue by kknd22** [gradle/gradle#3029]
`gradle dist` generates `playBinary `, which contains at line 165
```
eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $PLAY_BINARY_OPTS -classpath "\"$CLASSPATH\"" play.core.server.NettyServer "$APP_ARGS"
```  
which causes the following error when run  `playBinary `:  
`Could not find or load main class play.core.server.NettyServer    
`    
  
  
Looks like the correct class should be: `play.core.server.ProdServerStart`  

Should this just a be simple fix?
Thanks  

-cl

**Comment by bmuschko** [comment]
Is this potentially a duplicate of https://github.com/gradle/gradle/issues/2999?

**Comment by kknd22** [comment]
it is possible - but I checked 
https://github.com/playframework/playframework/commit/7bf7571c6997761baff79a0e89b08f9f5525b08f#diff-58bf6474fbdd7970aeb1630b24c9f7e2
  
and did not see the fix - maybe it should go in with the same fix?  

did and a quick search: 
https://github.com/gradle/gradle/search?utf8=%E2%9C%93&q=play.core.server.NettyServer&type=
  
shows only 3 files might need to be updated...

-cl

**Comment by marcospereira** [comment]
@bmuschko this does not look like a duplicate to me. #2999 is about the dev mode while this is related the prod server (and packaging the application).

@kknd22 after #3018 you won't need to change `org.gradle.play.internal.run.DefaultVersionedPlayRunAdapter` because in #3018 I'm overriding `runDevHttpServer` to adapt it to Play.

**Comment by kknd22** [comment]
@marcospereira what you said is correct for the  
`org.gradle.play.internal.run.DefaultVersionedPlayRunAdapter `   
  

looks like the other two files still need to be updated :
`org/gradle/play/plugins/PlayDistributionPluginTest.groovy`  and `org/gradle/play/plugins/PlayDistributionPlugin.java`

**Comment by marcospereira** [comment]
@kknd22 I'm not too familiar with Gradle code base, but shouldn't `org/gradle/play/plugins/PlayDistributionPlugin.java` be somehow extensible to support distinct versions of Play?

While using `play.core.server.NettyServer` won't work for Play 2.6, it could be ok to older versions.

WDYT?

**Comment by nberthet** [comment]
@kknd22 I noticed the issue as well, as a workaround, you can manually change the main class for the start script, in your build.gradle:
```
   tasks.whenTaskAdded {
        switch (it.name) {
        case "createPlayBinaryStartScripts":
            it.mainClassName = 'play.core.server.ProdServerStart'
            break
            // ...
        }
    }
```

**Comment by kknd22** [comment]
@marcospereira you are right - a new "switch for 2.6.x" needed for the 2.6 case but leave the thing as is for backward compatibility. Maybe a similar approach as `PlayRunAdapterV26X.java` ...

**Comment by kknd22** [comment]
@nberthet thanks I will give it a try....

**Comment by marcospereira** [comment]
@bmuschko any directions on how to approach this? I can submit a pull request later, but first I prefer to have a clear direction of how to solve this problem (since I'm familiar with Gradle code).

@nberthet nice workaround. 👍

**Comment by kknd22** [comment]
@bmuschko we are in mid of doing the gradle 4.2, play 2.6, scala 2.12 upgrade - if this patch works it would be great

**Comment by nberthet** [comment]
Same here, gradle 4.2, play 2.6, sticking to scala 2.11 due to dependencies, my projects are in kotlin anyway ;)

**Comment by bmuschko** [comment]
Thanks for chiming in on the conversation. We'll try to prioritize a fix soon. Will keep you in the loop.

**Comment by kknd22** [comment]
@marcospereira 
It works for the sample project now - will do more testing later. Thanks
